_posts/2019-11-09-using-vault-with-strimzi.md

```diff
@@ -0,0 +1,184 @@
+> The Cluster Operator automatically sets up TLS certificates to enable encryption and authentication within your cluster. It also sets up other TLS certificates if you want to enable encryption or TLS authentication between Kafka brokers and clients.
+
+By default, Strimzi generates its own CA and stores the CA certificate and private key as Kubernetes secrets. Using these two secrets, Strimzi generates certificates for all of the components in the cluster, which in turn ensures all communication is encrypted!
+
+## Bringing your own CA
+
+Now, in order to bring your own CA, you need to do this yourself and add the necessary certificate and key as Kubernetes secrets manually, adding the following labels: 
+
+```bash
+  strimzi.io/kind="Kafka"
+  strimzi.io/cluster="${clusterName}"
+```
+
+# Why Vault?
+
+Using Vault to create your own CA arguably gives you more control over your TLS configuration.
+
+Consider a Kafka cluster with a Kafka client you have developed. Your cluster requires that clients authenticate themselves using SSL, i.e. `ssl.client.auth="required"`.
+
+Using Vault as a CA, you can have an intermediate CA used for signing certificates for both your clients and brokers. Because Vault supports revocation, you can revoke any client certificates which are no longer trusted. This can be done simply, without much effort required. Consider a situation without Vault, fulfilling these tasks would be a lot more cumbersome! Indeed, the same can be done for your cluster certificates, but only by _revoking the intermediate CA itself._ More on this in the [Separate intermediate CA per cluster](#separate-intermediate-CA-per-cluster) section below.
+
+Vault integrates particularly well with Kubernetes, for example using the [Kubernetes Auth Method](https://www.vaultproject.io/docs/auth/kubernetes.html).
+
+See the Vault documentation for a full list of features:
+
+* [Secrets Engines](https://www.vaultproject.io/docs/secrets/index.html)
+
+* [Auth Methods](https://www.vaultproject.io/docs/auth/index.html)
+
+# Configuration of the Vault PKI secrets engine
+
+Assuming you have a Vault cluster running, you can [create your own CA by doing the following](https://learn.hashicorp.com/vault/secrets-management/sm-pki-engine):
+
+```bash
+# Creating the root CA:
+# First, enable the pki secrets engine at the pki path:
+$ vault secrets enable pki
+
+# Tune the pki secrets engine to issue certificates with a maximum time-to-live (TTL) 
+#   of 87600 hours.
+$ vault secrets tune -max-lease-ttl=87600h pki
+
+# Generate the root certificate and save the certificate in CA_cert.crt.
+$ vault write -field=certificate pki/root/generate/internal common_name="example.com" \
+       ttl=87600h > CA_cert.crt
+
+# This generates a new self-signed CA certificate and private key. Vault will automatically
```

Comment:
> This generates a new self-signed CA certificate and private key.

I assume does the `CA_cert.crt` just contains the certificate, and the key is kept within vault? It might be worth being explicit about that.

Reply:
@tombentley Yes, precisely. So in this case we never get to see the root's secret key.